### PR TITLE
Generating dsDNA with sticky ends (Example provided)

### PR DESCRIPTION
**Pull request by jinhzhou**
Modified generate-sa.py to support generation of dsDNA with sticky ends (Example provided)

### CODE REVIEW

**Comment by lorenzo-rovigatti** [comment]
Hello, and thanks for the PR, it's great to have people contributing to oxDNA.

I think this feature could be also useful to others, but I'd rather avoid the proliferation of scripts. Therefore, if you want to have it added to the codebase if would be best if you could modify generate-sa.py rather than create a new script.
In addition, the example folder should contain a README file that tells the user how to use it, and only files that are required to start the example should be committed. You can have a look at one of the newer examples (*e.g.* `PERSISTENCE_LENGTH`) to get the idea.

**Comment by jinhzhou** [comment]
Hi,

Thank you for the feedback! Hopefully I implemented them correctly! Let me know if there's anything else to fix up!

**Comment by lorenzo-rovigatti** [comment]
Thanks a lot, it looks much better. 

One last thing: could you delete the initial files that the user has to generate with `generate-sa.py`? In this way if something goes wrong in that stage the user and the user does not realise it, they won't be able to run the simulation.

File: utils/generate-sa.py [hunk 1]
```diff
@@ -215,24 +212,104 @@ def generate(self, bp, sequence=None, start_pos=np.array([0, 0, 0]), dir=np.arra
...
+    for i in range (1, len(strand) - 1):
+        print(strand_count, strand[i], nt_count - 1, nt_count + 1, file=output_file)
+        nt_count += 1
+    print(strand_count, strand[-1], nt_count - 1, -1, file=output_file)
+    nt_count += 1
+    strand_count += 1
+    return nt_count, strand_count
+
+
+# helper function to avoid code repeition in the read_strands() function
```
Comment (lorenzo-rovigatti) [inline utils/generate-sa.py:307]: "repeition" -> "repetition"

File: utils/generate-sa.py [hunk 2]
```diff
@@ -215,24 +212,104 @@ def generate(self, bp, sequence=None, start_pos=np.array([0, 0, 0]), dir=np.arra
...
+            mynewa3s.append (a3)
+            a1 = np.dot(R, a1)
+            rb += a3 * (BASE_BASE)
+
+        assert (len (mynewpositions) > 0)
+
+        return [mynewpositions, mynewa1s, mynewa3s]
+
+
+# helper function to avoid code repeition in the read_strands() function
```
Comment (lorenzo-rovigatti) [inline utils/generate-sa.py:294]: "repeition" -> "repetition"

**Comment by jinhzhou** [comment]
All files generated with generate-sa.py should be deleted now! Thank you for fixing the typo, silly me!

**Comment by lorenzo-rovigatti** [comment]
Sorry for being a bit pedantic, but there is also `energy.dat`, `last_conf.dat`, `last_conf.dat.pyidx` and `trajectory.dat`

**Comment by jinhzhou** [comment]
No worries! Those files have now been deleted.

**Pull request merged**